### DISCUSSION TITLE
[ActionView] Supercharge dom_id, allow deeply nested dom_ids and further prefix/suffix customization

### DISCUSSION THREAD

**Discussion by joelzwarrington** [rails/rails#53263] [Ideas]
I love [decomposing with turbo frames](https://turbo.hotwired.dev/handbook/frames), but find the current helper to get an ID difficult to use when nesting forms which might be duplicated on the page.

As an example...

Let's say we have a tag/tagging system and can edit tags for posts on a single page.

```erb
<% @posts.each do |post| %>
  <%= turbo_frame_tag(post) do %>
    <article>
      <h2><%= post.title %></h2>
      <%= turbo_frame_tag(dom_id(post, :tags)) do %>
        <%= render post.tags %>
      <% end %>
    <% end %>
  </article>
<% end %>
```

results in

```html
<turbo-frame id="post-1">
  <article>
    <h2>Lorem Ipsum</h2>
    <turbo-frame id="tags-post-1">
      <--! tags rendered here -->
    </turbo-frame>
  </article>
</turbo-frame>
```

I'd love for dom_id to work similarly to url_for, where you can pass as many positional arguments and it will concatenate it.

so doing
`dom_id(post, :tags)` would result in `post_1_tags`, doing `dom_id(:edit, post)` results in `edit_post_1`. also changing the behaviour for new unpersisted records, so it's always new_post or post_1, regardless if prefixes are passed.

In addition, you could pass multiple models, `dom_id(post, comment)` would produce  `post_1_comment_1`

**Comment by dvodvo** [comment]
Something that works:  `dom_id(post, dom_id(comment) )` 

I have had to stare down a case where three ids are necessary for a single dom_id.  Extending your example:
```
<% ci = dom_id(comment, item) %>
<%= turbo_frame_tag dom_id(post, ci) do %>
```
thus you can devise your own optional prefix to include limitless objects identified.
